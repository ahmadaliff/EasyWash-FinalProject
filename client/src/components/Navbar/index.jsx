import PropTypes from 'prop-types';
import { useState } from 'react';
import { FormattedMessage, injectIntl } from 'react-intl';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';

import Avatar from '@mui/material/Avatar';
import Menu from '@mui/material/Menu';
import MenuItem from '@mui/material/MenuItem';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import LightModeIcon from '@mui/icons-material/LightMode';
import NightsStayIcon from '@mui/icons-material/NightsStay';
import { Button } from '@mui/material';
import {
  DryCleaningOutlined,
  FavoriteBorder,
  Language,
  LibraryBooksOutlined,
  LocalLaundryServiceOutlined,
  LogoutOutlined,
  MessageOutlined,
  PersonOutline,
  ShoppingCartOutlined,
} from '@mui/icons-material';

import DialogLanguage from '@components/DialogLanguage';
import NavbarIconButton from '@components/NavbarIconButton';

import config from '@config/index';

import { setLocale, setTheme } from '@containers/App/actions';
import { actionHandleLogout } from '@containers/Client/actions';

import classes from './style.module.scss';

const Navbar = ({ locale, theme, user, login, intl: { formatMessage } }) => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const [openDialogLanguage, setOpenDialogLanguage] = useState(false);
  const [profilePosition, setProfilePosition] = useState(null);
  const openProfil = Boolean(profilePosition);

  const handleProfilClick = (event) => {
    setProfilePosition(event.currentTarget);
  };

  const handleProfileClose = () => {
    setProfilePosition(null);
  };

  const handleTheme = () => {
    dispatch(setTheme(theme === 'light' ? 'dark' : 'light'));
  };

  const onSelectLang = (e, lang) => {
    if (lang !== locale) {
      dispatch(setLocale(lang));
    }
    handleProfileClose();
    setOpenDialogLanguage(false);
  };

  const goHome = () => {
    navigate('/');
  };

  return (
    <div className={classes.headerWrapper} data-testid="navbar">
      <div className={classes.contentWrapper}>
        <div className={classes.logoImage} onClick={goHome}>
          <img src="/longLogo.svg" alt="logo" className={classes.logo} />
        </div>
        <div className={classes.toolbar}>
          {login && (
            <div className={classes.navButton}>
              {user?.role === 'user' && (
                <>
                  <NavbarIconButton title="Favorit" onClick={() => navigate('/favorit')} icon={<FavoriteBorder />} />
                  <NavbarIconButton
                    title={formatMessage({ id: 'app_cart_header' })}
                    onClick={() => navigate('/cart')}
                    icon={<ShoppingCartOutlined />}
                  />
                </>
              )}
              {user?.role === 'merchant' && (
                <>
                  <NavbarIconButton
                    title={formatMessage({ id: 'app_services_header' })}
                    onClick={() => navigate('/service')}
                    icon={<DryCleaningOutlined />}
                  />
                  <NavbarIconButton
                    title={formatMessage({ id: 'app_merchant_header' })}
                    onClick={() => navigate('/laundry')}
                    icon={<LocalLaundryServiceOutlined />}
                  />
                </>
              )}
              {user?.role !== 'admin' ? (
                <>
                  <NavbarIconButton
                    title={formatMessage({ id: 'app_order_list' })}
                    onClick={() => {
                      if (user?.role === 'user') navigate('/user/order');
                      else navigate('/laundry/orders');
                    }}
                    icon={<LibraryBooksOutlined />}
                  />
                  <NavbarIconButton
                    title={formatMessage({ id: 'app_chat' })}
                    onClick={() => navigate('/chat')}
                    icon={<MessageOutlined />}
                  />
                </>
              ) : (
                <NavbarIconButton
                  title={formatMessage({ id: 'app_user_page_header' })}
                  onClick={() => navigate('/admin/user')}
                  icon={<PersonOutline />}
                />
              )}
              <span className={classes.verticalLine} />
            </div>
          )}
          <div className={classes.toggle} onClick={handleProfilClick}>
            {login ? (
              <>
                {user?.imagePath ? (
                  <Avatar className={classes.avatar} src={`${config.api.server}${user?.imagePath}`} />
                ) : (
                  <Avatar className={classes.avatar}>
                    <p>
                      {user?.fullName?.split(' ')[0][0]}{' '}
                      {user?.fullName?.split(' ').length > 1 && user?.fullName?.split(' ')[1][0]}
                    </p>
                  </Avatar>
                )}
                <div className={classes.name}>{user?.fullName}</div>
              </>
            ) : (
              <Avatar className={classes.avatar} />
            )}
            <ExpandMoreIcon className={classes.arrow} />
          </div>
        </div>
        <Menu open={openProfil} anchorEl={profilePosition} onClose={handleProfileClose} className={classes.menu}>
          {login ? (
            <div>
              <MenuItem className={classes.menuItem} onClick={() => navigate('/profile')}>
                <Avatar className={classes.avatarMenuItem} />
                <FormattedMessage id="app_profile" />
              </MenuItem>
              <MenuItem className={classes.menuItem} onClick={() => dispatch(actionHandleLogout())}>
                <LogoutOutlined />
                <FormattedMessage id="app_header_logout" />
              </MenuItem>
            </div>
          ) : (
            <MenuItem className={classes.menuItem}>
              <Button
                variant="contained"
                className={classes.menuItemButtonContained}
                onClick={() => navigate('/login')}
              >
                <FormattedMessage id="app_header_login" />
              </Button>
              <Button
                variant="outlined"
                className={classes.menuItemButtonOutlined}
                onClick={() => navigate('/register')}
              >
                <FormattedMessage id="app_header_register" />
              </Button>
            </MenuItem>
          )}
          <hr />
          <MenuItem onClick={() => setOpenDialogLanguage(true)} className={classes.menuItem}>
            <Language />
            <p>
              <FormattedMessage id="app_select_language" />
            </p>
          </MenuItem>
          <MenuItem onClick={handleTheme} className={`${classes.menuItem} ${classes.theme}`} data-testid="toggleTheme">
            {theme === 'light' ? <NightsStayIcon /> : <LightModeIcon />}
            <p>{theme}</p>
          </MenuItem>
        </Menu>
        <DialogLanguage
          handleSelectLang={onSelectLang}
          handleClose={() => setOpenDialogLanguage(false)}
          open={openDialogLanguage}
          locale={locale}
        />
      </div>
    </div>
  );
};

Navbar.propTypes = {
  locale: PropTypes.string.isRequired,
  theme: PropTypes.string,
  user: PropTypes.object,
  intl: PropTypes.object,
  login: PropTypes.bool,
};

export default injectIntl(Navbar);
