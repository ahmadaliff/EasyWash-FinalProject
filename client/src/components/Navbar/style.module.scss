@import '@styles/base';

.headerWrapper {
  background-color: var(--color-bg) !important;
  border-bottom: 1px solid rgba($color: gray, $alpha: 0.2);
  position: sticky;
  top: 0;
  z-index: 15;
  .contentWrapper {
    @include directionX;

    color: var(--color-text);
    padding: 0.75rem;
    align-items: center;
    justify-content: space-between;
    font-size: 1rem;

    .logoImage {
      @include directionX;
      cursor: pointer;
      align-items: center;
      gap: 0.5rem;
      .logo {
        width: auto;
        height: 2rem;
        &:hover {
          transform: scale(1.05, 1.05);
        }
      }
    }

    .toolbar {
      @include directionX;
      align-items: center;
      .navButton {
        display: none;
        @include apply-at-sm-size {
          @include directionX;
        }
        gap: 0.25rem;
        color: var(--color-text);
        svg {
          height: 1.25rem;
          width: 1.25rem;
        }
      }
      .verticalLine {
        display: none;
        @include apply-at-sm-size {
          display: block;
        }
        height: 1.5rem;
        width: 2px;
        margin: 0 1.25rem;
        background-color: rgba($color: gray, $alpha: 0.75);
      }
      .toggle {
        @include directionX;
        align-items: center;
        cursor: pointer;
        gap: 0.125rem;
        .avatar {
          width: 1.75rem;
          height: 1.75rem;
          box-shadow: rgba(99, 99, 99, 0.2) 0px 1px 10px 0px;
          margin-right: 0.25rem;
          background-color: var(--color-primary);
          p {
            text-transform: uppercase;
            font-size: small;
            font-weight: 700;
            color: white;
          }
        }

        .name {
          font-family: Poppins, Arial, Helvetica, sans-serif;
          font-weight: 700;
          font-size: 0.75rem;
          text-transform: capitalize;
        }

        .arrow {
          font-size: 1.125rem;
        }
      }
    }
  }
}

.menu {
  @include directionX;
  align-items: center;

  .menuItem {
    font-family: Poppins, Arial, Helvetica, sans-serif;
    font-size: small;
    text-transform: capitalize;
    gap: 0.5rem;
    .avatarMenuItem {
      width: 1.5rem;
      height: 1.5rem;
    }
    .menuItemButtonContained,
    .menuItemButtonOutlined {
      text-transform: capitalize;
      font-size: small;
      padding: 0.45rem 2rem;
    }
    .menuItemButtonContained {
      background-color: var(--color-primary);
    }
    .menuItemButtonOutlined {
      color: var(--color-primary);
      border-color: var(--color-primary);
    }
    p {
      margin: 0;
    }
    svg {
      width: 1.25rem;
      height: 1.25rem;
    }
  }
  .theme {
    @include directionX;
    cursor: pointer;
  }
}
