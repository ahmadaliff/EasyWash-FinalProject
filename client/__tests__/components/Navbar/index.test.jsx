import { fireEvent, render, screen } from '@utils/testHelper';
import Navbar from '@components/Navbar';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  useNavigate: () => mockNavigate,
}));

let navbarUser;
let navbarAdmin;
let navbarMerchant;
let navbarWithoutLogin;

const mockProps = {
  locale: 'id',
  theme: 'light',
  login: true,
  intl: { formatMessage: jest.fn() },
};

describe('Navbar Component', () => {
  describe('Navbar with role User', () => {
    beforeEach(() => {
      navbarUser = render(
        <Navbar
          {...mockProps}
          user={{
            role: 'user',
            fullName: 'ahmad alif sofian ',
            iamgePath: null,
          }}
        />
      );
    });
    test('Correct render', () => {
      const { getByTestId } = navbarUser;
      expect(getByTestId('navbar')).toBeInTheDocument();
    });
    test('Should be render text', () => {
      expect(screen.getByText('ahmad alif sofian')).toBeInTheDocument();
    });
    test('Should call navigate to profile when button clicked', () => {
      const { getByTestId } = navbarUser;
      const togleButton = getByTestId('button-togle');
      fireEvent.click(togleButton);
      const button = getByTestId('button-profile');
      fireEvent.click(button);
      expect(mockNavigate).toHaveBeenCalledWith(`/profile`);
    });
    test('Should call navigate to favorit when button clicked', () => {
      const { getByTestId } = navbarUser;
      const button = getByTestId('button-favorit');
      fireEvent.click(button);
      expect(mockNavigate).toHaveBeenCalledWith(`/favorit`);
    });
    test('Should call navigate to cart when button clicked', () => {
      const { getByTestId } = navbarUser;
      const button = getByTestId('button-cart');
      fireEvent.click(button);
      expect(mockNavigate).toHaveBeenCalledWith(`/cart`);
    });
    test('Should call navigate to order when button clicked', () => {
      const { getByTestId } = navbarUser;
      const button = getByTestId('button-order');
      fireEvent.click(button);
      expect(mockNavigate).toHaveBeenCalledWith(`/user/order`);
    });
    test('Should call navigate to chat when button clicked', () => {
      const { getByTestId } = navbarUser;
      const button = getByTestId('button-chat');
      fireEvent.click(button);
      expect(mockNavigate).toHaveBeenCalledWith(`/chat`);
    });
  });
  describe('Navbar with role Admin', () => {
    beforeEach(() => {
      navbarAdmin = render(
        <Navbar
          {...mockProps}
          user={{
            role: 'admin',
            fullName: 'ahmad alif sofian ',
            iamgePath: null,
          }}
        />
      );
    });
    test('Correct render', () => {
      const { getByTestId } = navbarAdmin;
      expect(getByTestId('navbar')).toBeInTheDocument();
    });
    test('Should be render text', () => {
      expect(screen.getByText('ahmad alif sofian')).toBeInTheDocument();
    });
    test('Should call navigate to admin iuser when button clicked', () => {
      const { getByTestId } = navbarAdmin;
      const button = getByTestId('button-admin');
      fireEvent.click(button);
      expect(mockNavigate).toHaveBeenCalledWith(`/admin/user`);
    });
    test('Should match with snapshot', () => {
      expect(navbarAdmin).toMatchSnapshot();
    });
  });
  describe('Navbar with role Merchant', () => {
    beforeEach(() => {
      navbarMerchant = render(
        <Navbar
          {...mockProps}
          user={{
            role: 'merchant',
            fullName: 'ahmad alif sofian ',
            iamgePath: null,
          }}
        />
      );
    });
    test('Correct render', () => {
      const { getByTestId } = navbarMerchant;
      expect(getByTestId('navbar')).toBeInTheDocument();
    });
    test('Should be render text', () => {
      expect(screen.getByText('ahmad alif sofian')).toBeInTheDocument();
    });
    test('Should call navigate to service when button clicked', () => {
      const { getByTestId } = navbarMerchant;
      const button = getByTestId('button-service');
      fireEvent.click(button);
      expect(mockNavigate).toHaveBeenCalledWith(`/service`);
    });
    test('Should call navigate to laundry when button clicked', () => {
      const { getByTestId } = navbarMerchant;
      const button = getByTestId('button-laundry');
      fireEvent.click(button);
      expect(mockNavigate).toHaveBeenCalledWith(`/laundry`);
    });
  });
  describe('Navbar without login', () => {
    beforeEach(() => {
      navbarWithoutLogin = render(<Navbar {...mockProps} user={undefined} login={false} />);
    });
    test('should Correct render', () => {
      const { getByTestId } = navbarWithoutLogin;
      expect(getByTestId('navbar')).toBeInTheDocument();
    });
    test('Should call navigate to login when button clicked', () => {
      const { getByTestId } = navbarWithoutLogin;
      const togleButton = getByTestId('button-togle');
      fireEvent.click(togleButton);
      const button = getByTestId('button-login');
      fireEvent.click(button);
      expect(mockNavigate).toHaveBeenCalledWith(`/login`);
    });
    test('Should call navigate to register when button clicked', () => {
      const { getByTestId } = navbarWithoutLogin;
      const togleButton = getByTestId('button-togle');
      fireEvent.click(togleButton);
      const button = getByTestId('button-register');
      fireEvent.click(button);
      expect(mockNavigate).toHaveBeenCalledWith(`/register`);
    });
    test('Should call open dialog language when button clicked', () => {
      const { getByTestId } = navbarWithoutLogin;
      const togleButton = getByTestId('button-togle');
      fireEvent.click(togleButton);
      const buttonDialog = getByTestId('button-dialog-language');
      fireEvent.click(buttonDialog);
    });
    test('Should change theme when button clicked', () => {
      const { getByTestId } = navbarWithoutLogin;
      const togleButton = getByTestId('button-togle');
      fireEvent.click(togleButton);
      const buttonTheme = getByTestId('toggleTheme');
      fireEvent.click(buttonTheme);
      expect(screen.getByText('light')).toBeInTheDocument();
    });
    test('Should match with snapshot', () => {
      expect(navbarMerchant).toMatchSnapshot();
    });
  });
});
