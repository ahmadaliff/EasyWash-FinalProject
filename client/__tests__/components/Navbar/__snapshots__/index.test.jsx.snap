// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component Navbar with role Admin Should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style=""
  >
    <div>
      <div
        class="headerWrapper"
        data-testid="navbar"
      >
        <div
          class="contentWrapper"
        >
          <div
            class="logoImage"
          >
            <img
              alt="logo"
              class="logo"
              src="/longLogo.svg"
            />
          </div>
          <div
            class="toolbar"
          >
            <div
              class="navButton"
            >
              <button
                aria-label="Manajemen User"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                data-mui-internal-clone-element="true"
                data-testid="button-admin"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="PersonOutlineIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-label="Laundry yang dihapus"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                data-mui-internal-clone-element="true"
                data-testid="button-deletedMerchant"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="LayersClearIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m19.81 14.99 1.19-.92-1.43-1.43-1.19.92 1.43 1.43zm-.45-4.72L21 9l-9-7-2.91 2.27 7.87 7.88 2.4-1.88zM3.27 1 2 2.27l4.22 4.22L3 9l1.63 1.27L12 16l2.1-1.63 1.43 1.43L12 18.54l-7.37-5.73L3 14.07l9 7 4.95-3.85L20.73 21 22 19.73 3.27 1z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <span
                class="verticalLine"
              />
            </div>
            <div
              class="toggle"
              data-testid="button-togle"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault avatar css-2s90m6-MuiAvatar-root"
              >
                <p>
                  a
                   
                  a
                </p>
              </div>
              <div
                class="name"
              >
                ahmad alif sofian 
              </div>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium arrow css-i4bv87-MuiSvgIcon-root"
                data-testid="ExpandMoreIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="headerWrapper"
      data-testid="navbar"
    >
      <div
        class="contentWrapper"
      >
        <div
          class="logoImage"
        >
          <img
            alt="logo"
            class="logo"
            src="/longLogo.svg"
          />
        </div>
        <div
          class="toolbar"
        >
          <div
            class="navButton"
          >
            <button
              aria-label="Manajemen User"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              data-mui-internal-clone-element="true"
              data-testid="button-admin"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="PersonOutlineIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="Laundry yang dihapus"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              data-mui-internal-clone-element="true"
              data-testid="button-deletedMerchant"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="LayersClearIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m19.81 14.99 1.19-.92-1.43-1.43-1.19.92 1.43 1.43zm-.45-4.72L21 9l-9-7-2.91 2.27 7.87 7.88 2.4-1.88zM3.27 1 2 2.27l4.22 4.22L3 9l1.63 1.27L12 16l2.1-1.63 1.43 1.43L12 18.54l-7.37-5.73L3 14.07l9 7 4.95-3.85L20.73 21 22 19.73 3.27 1z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <span
              class="verticalLine"
            />
          </div>
          <div
            class="toggle"
            data-testid="button-togle"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault avatar css-2s90m6-MuiAvatar-root"
            >
              <p>
                a
                 
                a
              </p>
            </div>
            <div
              class="name"
            >
              ahmad alif sofian 
            </div>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium arrow css-i4bv87-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Navbar Component Navbar without login Should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style=""
  >
    <div>
      <div
        class="headerWrapper"
        data-testid="navbar"
      >
        <div
          class="contentWrapper"
        >
          <div
            class="logoImage"
          >
            <img
              alt="logo"
              class="logo"
              src="/longLogo.svg"
            />
          </div>
          <div
            class="toolbar"
          >
            <div
              class="toggle"
              data-testid="button-togle"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault avatar css-2s90m6-MuiAvatar-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
                  data-testid="PersonIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                  />
                </svg>
              </div>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium arrow css-i4bv87-MuiSvgIcon-root"
                data-testid="ExpandMoreIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
